handlers/user.go

```diff
@@ -2078,7 +2079,43 @@ func isValidSubscriptionFilter(eventName types.EventName, filter string) bool {
 		isClient = true
 	}
 
-	return len(filter) == 0 || isPkey || isClient
+	isValidMachine := false
+	if types.IsMachineNotification(eventName) {
+		machines, err := db.BigtableClient.GetMachineMetricsMachineNames(userID)
+		if err != nil {
+			return false, errors.Wrap(err, "can not get users machines from bigtable for validation")
+		}
+		for _, userMachineName := range machines {
+			if userMachineName == filter {
+				isValidMachine = true
+				break
+			}
+		}
+
+		// While the above works fine for active machines (adding a new notification to an active machine)
+		// It does not work for a machine that is offline and where the user wants to unsubscribe from this machine.
+		// So check the db for any machine names as well
+		if !isValidMachine {
+			machines := make([]string, 0)
+			err = db.FrontendWriterDB.Select(&machines, `
+				select event_filter
+				from users_subscriptions 
```

Comment:
**Question**: every instance where `isValidSubscriptionFilter` is called may cause inserting into the `users_subscriptions` table, right? If a user subscribes to his machine for the first time and has no filters to check against in the DB yet, this validation would fail. Can this happen? Or does having a machine automatically mean you have at least one subscription to it.